src/native/managed/cdacreader/tests/MockDescriptors/MockDescriptors.HashMap.cs

```diff
@@ -50,38 +51,42 @@ public HashMap(MockMemorySpace.Builder builder)
 
         public HashMap(MockMemorySpace.Builder builder, (ulong Start, ulong End) allocationRange)
         {
-            _builder = builder;
-            _allocator = _builder.CreateAllocator(allocationRange.Start, allocationRange.End);
-            Types = GetTypes();
-            Globals =
-            [
-                (nameof(Constants.Globals.HashMapSlotsPerBucket), HashMapSlotsPerBucket, "uint32"),
-                (nameof(Constants.Globals.HashMapValueMask), _builder.TargetTestHelpers.PointerSize == 4 ? 0x7FFFFFFFu : 0x7FFFFFFFFFFFFFFFu, "uint64"),
-            ];
+            Builder = builder;
+            _allocator = Builder.CreateAllocator(allocationRange.Start, allocationRange.End);
+            Types = GetTypes(builder.TargetTestHelpers);
+            Globals = GetGlobals(builder.TargetTestHelpers);
         }
 
-        private Dictionary<DataType, Target.TypeInfo> GetTypes()
+        internal static Dictionary<DataType, Target.TypeInfo> GetTypes(TargetTestHelpers helpers)
         {
             return GetTypesForTypeFields(
-                _builder.TargetTestHelpers,
+                helpers,
                 [
                     HashMapFields,
-                    BucketFields(_builder.TargetTestHelpers),
+                    BucketFields(helpers),
                 ]);
         }
 
+        internal static (string Name, ulong Value)[] GetGlobals(TargetTestHelpers helpers)
+        {
+            return [
+                (nameof(Constants.Globals.HashMapSlotsPerBucket), HashMapSlotsPerBucket),
+                (nameof(Constants.Globals.HashMapValueMask), helpers.PointerSize == 4 ? 0x7FFFFFFFu : 0x7FFFFFFFFFFFFFFFu),
```

Comment:
> `helpers.PointerSize == 4 ? 0x7FFFFFFFu : 0x7FFFFFFFFFFFFFFFu`

Should we have a member that returns `MaxSignedTargetAddress`? I think the cDAC reader logic should have that too.